### PR TITLE
Implement the commerce settings connect URL method

### PR DESCRIPTION
**Pull request by ChaseWiseman**
# Summary

Builds the Commerce connect URL.

## UI Changes

_N/A_

## QA

- `integration Admin/Settings_Screens` tests pass

We will confirm further with acceptance tests once the settings screen has been implemented.

## Before merge

- [ ] I have confirmed these changes in each supported minor WooCommerce version

### CODE REVIEW

File: includes/Admin/Settings_Screens/Commerce.php [hunk 1]
```diff
@@ -83,7 +84,29 @@ public function render_google_product_category_field() {
...
 
-		return '';
+		// build the site URL to which the user will ultimately return
+		$site_url = add_query_arg( [
+			'wc-api'               => Connection_Handler::ACTION_CONNECT_COMMERCE,
+			'nonce'                => wp_create_nonce( Connection_Handler::ACTION_CONNECT_COMMERCE ),
+		], home_url( '/' ) );
+
+		// build the proxy app URL where the user will land after onboarding, to be redirected to the site URL
+		$redirect_url = add_query_arg( 'site_url', $site_url, facebook_for_woocommerce()->get_connection_handler()->get_proxy_url() );
```
Comment (wvega) [inline includes/Admin/Settings_Screens/Commerce.php:107]: I think we should encode `$site_url` as well:

````suggestion
		$redirect_url = add_query_arg( 'site_url', urlencode( $site_url ), facebook_for_woocommerce()->get_connection_handler()->get_proxy_url() );
```

File: tests/integration/Admin/Settings_Screens/CommerceTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Tests\Admin\Settings_Screens;
+
+use SkyVerge\WooCommerce\Facebook\Admin\Settings_Screens;
+
+/**
+ * Tests the Admin\Settings class.
```
Comment (wvega) [inline tests/integration/Admin/Settings_Screens/CommerceTest.php:8]:
```suggestion
 * Tests the Settings_Screens\Commerce class.
```

File: tests/integration/Admin/Settings_Screens/CommerceTest.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	/** @see Settings_Screens\Commerce::get_connect_url() */
+	public function test_get_connect_url() {
+
+		$screen = new Settings_Screens\Commerce();
+		$connect_url = $screen->get_connect_url();
+
+		$this->assertStringContainsString( 'https://www.facebook.com/commerce_manager/onboarding/?app_id=', $connect_url );
+		$this->assertStringContainsString( 'redirect_url=https%3A%2F%2Fconnect.woocommerce.com%2Fauth%2Ffacebook%2F%3Fsite_url%3D', $connect_url );
+		$this->assertStringContainsString( 'wc-api%3D' . \SkyVerge\WooCommerce\Facebook\Handlers\Connection::ACTION_CONNECT_COMMERCE . '%26nonce%3D', $connect_url );
```
Comment (wvega) [inline tests/integration/Admin/Settings_Screens/CommerceTest.php:46]: If we choose to encode `$site_url` as well:

```suggestion
		$this->assertStringContainsString( 'wc-api%253D' . \SkyVerge\WooCommerce\Facebook\Handlers\Connection::ACTION_CONNECT_COMMERCE . '%2526nonce%253D', $connect_url );
```

**Review by wvega — CHANGES_REQUESTED** [review]
Looks good @ChaseWiseman. Only one change request before we merge:

**Comment by ChaseWiseman** [comment]
@wvega makes sense! thanks for the suggestions. applying and merging 👍

**Pull request merged**